tests/unit/commands/init/test_cli.py

```diff
@@ -224,6 +224,27 @@ def test_init_fails_invalid_dep_mgr(self, git_repo_clone_mock):
                 extra_context=None,
             )
 
+    @patch("samcli.lib.utils.git_repo.GitRepo.clone")
+    def test_init_fails_wrong_dep_mgr_for_runtime(self, git_repo_clone_mock):
+        # WHEN the wrong dependency_manager is passed for a runtime
+        # THEN an exception should be raised
+        with self.assertRaises(UserException):
```

Comment:
Isn't this a `InvalidInitOptionException`?
We may also want to check the message of the exception as it is a generic exception for init options
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises

Reply:
`InvalidInitOptionException` is of type `UserException`. I realized that other test used this and as such for consistency I used it, since `InvalidInitOptionException` is of type `UserException`

Reply:
If another code path raises an exception inheriting `UserException`, your test would still pass so you should expect an `InvalidInitOptionException`.